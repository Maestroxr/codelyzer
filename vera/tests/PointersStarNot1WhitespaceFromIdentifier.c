
int main()
{

	int x = 0;
	
	/*OK*/
	int* p1 = NULL;
	int* p2 = (int*)malloc(sizeof(int));
	int** p3 = (int**)malloc(sizeof(int)),***** p4 = (int*****)malloc(sizeof(int));
	/*--------*/
	
	/*PROBLEM*/
	int*  problem1 = NULL;
	int* pp2 = NULL,*problem2 = NULL;
	int**     p3 = (int**)malloc(sizeof(int)),*****            p4 = (int*****)malloc(sizeof(int));

	/*********/
	return x;
}



