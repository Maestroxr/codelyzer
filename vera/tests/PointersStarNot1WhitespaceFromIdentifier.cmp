C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNot1WhitespaceFromIdentifier.c:14: Pointer's '*' does not have a single whitespace before identifier, 'int* problem1' instead of 'int*  problem1'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNot1WhitespaceFromIdentifier.c:15: Pointer's '*' does not have a single whitespace before identifier, 'int ,* problem2' instead of 'int ,*problem2'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNot1WhitespaceFromIdentifier.c:16: Pointer's '*' does not have a single whitespace before identifier, 'int ,***** problem4' instead of 'int ,*****            problem4'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNot1WhitespaceFromIdentifier.c:16: Pointer's '*' does not have a single whitespace before identifier, 'int** problem3' instead of 'int**     problem3'.
