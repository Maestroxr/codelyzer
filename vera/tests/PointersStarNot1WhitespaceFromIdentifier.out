C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNot1WhitespaceFromIdentifier.c:9: Pointer's 'p2' Malloc(s) missing '1' null pointer check(s).
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNot1WhitespaceFromIdentifier.c:10: Malloc's cast is to incorrect type, pointer initialized as 'int**********' but cast was to 'int*****'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNot1WhitespaceFromIdentifier.c:14: Pointer's '*' does not have a single whitespace before identifier, 'int* problem1' instead of 'int*  problem1'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNot1WhitespaceFromIdentifier.c:15: Pointer's '*' does not have a single whitespace before identifier, 'int ,* problem2' instead of 'int ,*problem2'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNot1WhitespaceFromIdentifier.c:16: Malloc's cast is to incorrect type, pointer initialized as 'int**********' but cast was to 'int*****'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNot1WhitespaceFromIdentifier.c:16: Pointer's 'p3' Malloc(s) missing '4' null pointer check(s).
